sds_data_manager/lambda_code/SDSCode/dependency_config_ultra.py

```diff
@@ -16,80 +16,142 @@
+        dependent_descriptor="45all",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
     PreProcessingDependency(
         primary_instrument="ultra",
-        primary_data_level="l0",
-        primary_descriptor="raw",
+        primary_data_level="l1a",
+        primary_descriptor="45rates",
         dependent_instrument="ultra",
-        dependent_data_level="l1a",
-        dependent_descriptor="45phxtof",
+        dependent_data_level="l1b",
+        dependent_descriptor="45extended-spin",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
+    # TODO: uncomment for post-SIT-3
+    # PreProcessingDependency(
+    #     primary_instrument="codice",
+    #     primary_data_level="l2",
+    #     primary_descriptor="todo",
+    #     dependent_instrument="ultra",
+    #     dependent_data_level="l1b",
+    #     dependent_descriptor="45extended-spin",
+    #     relationship="HARD",
+    #     direction="DOWNSTREAM",
+    # ),
+    # PreProcessingDependency(
+    #     primary_instrument="swapi",
+    #     primary_data_level="l2",
+    #     primary_descriptor="todo",
+    #     dependent_instrument="ultra",
+    #     dependent_data_level="l1b",
+    #     dependent_descriptor="45extended-spin",
+    #     relationship="HARD",
+    #     direction="DOWNSTREAM",
+    # ),
     PreProcessingDependency(
         primary_instrument="ultra",
-        primary_data_level="l0",
-        primary_descriptor="raw",
+        primary_data_level="l1b",
+        primary_descriptor="45extended-spin",
         dependent_instrument="ultra",
-        dependent_data_level="l1a",
-        dependent_descriptor="45aux",
+        dependent_data_level="l1b",
+        dependent_descriptor="45culling-mask",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
     PreProcessingDependency(
         primary_instrument="ultra",
-        primary_data_level="l0",
-        primary_descriptor="raw",
+        primary_data_level="l1b",
+        primary_descriptor="45culling-mask",
         dependent_instrument="ultra",
-        dependent_data_level="l1a",
-        dependent_descriptor="45rates",
+        dependent_data_level="l1b",
+        dependent_descriptor="45badtimes",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
     PreProcessingDependency(
         primary_instrument="ultra",
-        primary_data_level="l0",
-        primary_descriptor="raw",
+        primary_data_level="l1a",
+        primary_descriptor="45de",
         dependent_instrument="ultra",
-        dependent_data_level="l1a",
-        dependent_descriptor="90events",
+        dependent_data_level="l1b",
+        dependent_descriptor="45annotated-de",
         relationship="HARD",
         direction="DOWNSTREAM",
     ),
     PreProcessingDependency(
         primary_instrument="ultra",
         primary_data_level="l0",
-        primary_descriptor="raw",
+        primary_descriptor="90raw",
```

Comment:
With above changes, you don’t need this dependency. Don’t forget to make same change to upstream as well for this.  
```suggestion

```